### ISSUE TITLE
[UI] ui/app/models/service.js generates broken proxy URL links for service endpoints of cluster services

### ISSUE THREAD

**Issue by TheDiveO** [rancher/rancher#16463]
**What kind of request is this (question/bug/enhancement/feature request):**
bug

**Steps to reproduce (least amount of steps as possible):**

1. create a `Service` resource with two HTTP ports (which need to have different names, such as "http" and "dashboard"):
```
apiVersion: v1
kind: Service
metadata:
  labels:
    app: foo
    io.cattle.field/appId: foo
    kubernetes.io/cluster-service: "true"
  name: foo
  namespace: default
spec:
  ports:
  - name: dashboard
    port: 8080
    protocol: TCP
    targetPort: 8080
  - name: http
    port: 5000
    protocol: TCP
    targetPort: 5000
  selector:
    app: foo
  sessionAffinity: None
  type: ClusterIP
status:
  loadBalancer: {}
```
2. Deploy the service as part of a catalog workload.

**Result:**
The workload UI now shows two hyperlinks for this workload, both named "/index.html" yet linking to different proxy URLs:
1. `https://xxx/k8s/clusters/c-xxxxx/api/v1/namespaces/default/services/dashboard:foo:8080/proxy/` -- this link does **not** work and results in:
```
{
  "kind": "Status",
  "apiVersion": "v1",
  "metadata": {
    
  },
  "status": "Failure",
  "message": "invalid service request \"dashboard:foo:8080\"",
  "reason": "BadRequest",
  "code": 400
}
```
2. `https://xxx/k8s/clusters/c-xxxxx/api/v1/namespaces/default/services/http:foo:5000/proxy/` -- this link works as expected.

The problem is with https://github.com/rancher/ui/blob/b585f8b0e73bc2520ea991bf7f9a09d784430b13/app/models/service.js#L32, which uses the port "name" as the first element before the service name. Instead, this first element must be either "http", "https", or simply left out. But it must not an arbitrary service endpoint name.

As the port name is used, it is impossible to use two ports named "http" to work around this bug.

Note bene: both ports are HTTP, because they are behind an authenticating proxy using only HTTPS.

**Environment information**
- Rancher version 2.1.1
- Installation option: single install:

**Comment by loganhz** [comment]
@TheDiveO Thanks for the report and solution!

It will be fixed in https://github.com/rancher/ui/pull/2379

**Comment by davidnuzik** [comment]
Version: master 2.2 (1/9/19)
Confirmed fix. I can no longer reproduce.
- Confirm URL pattern is correct. I used Kubernetes Dashboard and made sure the URL pattern for /index.html is correct.
- Fork the chart and modify svc.yaml and change the port to something else. Confirm this is working as expected for the url. Do the same and try with a snippet from the service yaml provided. Ensure the URL pattern is what we are expecting.
